streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -418,12 +418,20 @@ public synchronized void clean() {
         }
 
         try {
+            Utils.delete(stateDir);
```

Comment:
It seems to course grained to just call `Utils.delete`? In the end it's just a single file `PROCESS_FILE_NAME`.

We might also want to put this cleanup into it's own try-catch-block. We can either delete only the `PROCESS_FILE_NAME` or _all_ content of the state directory (might be more future proof). And only afterwards try to delete the _empty_ state directory itself.

Can we also extend the corresponding unit tests?